### ISSUE TITLE
Magus/Engineer pet/turret abilities

### ISSUE THREAD

**Issue by ElMosquito** [WarEmu/WarBugs#4953]
When I played as a magus in T1, I noticed something with the horrors that I did not see on the live server. When they are using their special abilities (like Daemonic Fire for pink horror for example) they take some time to cast them and freeze in their movement for like two seconds before sending away the abilities towards the target.
At first I thought it was just a visual bug that was not that big of a deal, but it just occured to me that the pets did not have any cast times for their abilities on live. It used to be a thing on the live server that if you had your pet on aggressive while it died or despawned, and then summoned it near enemy players/mobs, it would cast one of its abilities instantly on a nearby enemy and draw aggression from mobs (which you had to watch out for in pve).
I checked the abilities on each magus pet and noticed that some abilities now have cast times, which would explain why they freezed in their movements as they were not supposed to take that long to cast an ability. The same goes for engineer as far as I have been able to check.

Both of my characters are low level so I have not been able to confirm cast times for all pet/turret abilities (perhaps someone who has a high level magus/engi can check the remaining abilities). But I have been able to check that all horror/turret abilities are supposed to be instant cast, according to the descriptions in the Ability Training window (the pop up window you get at a trainer if clicking on Buy Career Abilities).
![abilitymagus](https://cloud.githubusercontent.com/assets/16590792/13821352/855e878e-eba1-11e5-9ca7-874faa1bf46d.jpg)
![abilityengi](https://cloud.githubusercontent.com/assets/16590792/13821355/8923c992-eba1-11e5-806a-2cbd2a210e03.jpg)

These abilities do not disappear from the training options as you level up, and my guess to why is that you are not actually getting those abilities for your horrors/turrets. You get completely different abilities with cast times added to some of them, which you can see when looking at the description of the abilities in the pet command icon.
![maguspet](https://cloud.githubusercontent.com/assets/16590792/13821945/15c312ac-eba4-11e5-866e-86b2924290ff.jpg)

As you can see for magus: Daemonic Fire, Warping Energy and Flames of Change have cast times of 3, 2 and 1 seconds while Flame of Tzeentch is instant cast as it should be. (Coruscating Energy and Daemonic Consumption are abilities I have not been able to check due to the low level of my magus.)

The same goes for engineer: Penetrating Round, Flamethrower and Shock Grenade all have 2 seconds cast time. (High-Explosive Grenade, Steam Vent and Machine Gun are abilities I have not been able to check yet since my engi is too low level.)
![turrets](https://cloud.githubusercontent.com/assets/16590792/13822284/54dc6712-eba5-11e5-8e16-14db777d5ef1.jpg)

And a last thing to report about horrors and turrets is that they never despawn unless you actively remove them, they are killed. They used to have a few minutes duration before you had to summon a new one or if you got too far away from it you had to summon a new one as well.

**Comment by Renork** [comment]
I don't normally pay attention to the pet since it's so shitty to begin with but you are correct, they did not have a casting animation on live. Here's a video https://www.youtube.com/watch?v=TJYFl3F11Mg

The pet here sometimes takes a while to begin casting, then it "loads" and stops for two second before firing finalizing the cast.

**Comment by Morksezstop** [comment]
turrets and magus horrors do dispell when you go too far, but for sure they dont have internal cd here on ror.

**Comment by Renork** [comment]
Point is the cast times, not the buggy resummon skill.

**Comment by genisaurus** [comment]
tl;dr: Pet abilities have cast times that they shouldn't. Because tooltips are clientside, it could be that they had cast times at some point during live, and they were "removed" by replacing them with copies that were instant-cast.

10/10 report, please report again.

**Comment by ozarmidrashim** [comment]
looked through this and it's the biggest joke I've yet seen - one ability appears to cast another ability for no apparent reason, and in the case of channels, those abilities intersect with one another.

Going to have to defer this a while

**Comment by ozarmidrashim** [comment]
Dealt with in recent rework
